server/src/test/java/org/elasticsearch/cluster/metadata/TemplateUpgradeServiceTests.java

```diff
@@ -90,7 +86,7 @@ public void testCalculateChangesAddChangeAndDelete() {
             IndexTemplateMetaData.builder("changed_test_template").patterns(randomIndexPatterns()).build()
         );
 
-        TemplateUpgradeService service = new TemplateUpgradeService(Settings.EMPTY, null, clusterService, null,
+        TemplateUpgradeService service = new TemplateUpgradeService(Settings.EMPTY, null, mock(ClusterService.class), null,
```

Comment:
I wonder if it would be better to use `ClusterServiceUtils#createClusterService` and other utility methods here. It reduces the mocks that we use

Reply:
@jaymode I got rid of `mock(ClusterService.class)` and `mock(ThreadPool.class)` inside `TemplateUpgradeServiceTests`. The client mocks survived (the only remaining mocks) since this is only an `ESTestCase` test suite.

The `TemplateUpgradeServiceTests#testClusterStateUpdate` relied on the mock threadpool to be synchronous so the test got a bit cluttered by some semaphores. The test is now tighter, as it checks the `tryFinishUpdate` and `calculateTemplateChanges` call counts too.

Can you please take another glance at it?